keps/sig-network/2091-cluster-network-policy/README.md

```diff
@@ -0,0 +1,958 @@
+will not be blocked, if developers create NetworkPolicy which isolates the Pods in
+tenant Namespaces. When there's a ClusterNetworkPolicy like `ingress-egress-gateway`
+present in the cluster, the above policy will be overridden as `Deny` rules have
+higher precedence than `Allow` rules. In that case, the `app=system` Namespaces need
+to be added to the Deny `except` list of `ingress-egress-gateway`.
+
+#### Story 5: Restrict egress to well known destinations
+
+```yaml
+apiVersion: networking.k8s.io/v1alpha1
+kind: ClusterNetworkPolicy
+metadata:
+  name: restrict-egress-to-db
+spec:
+  appliedTo:
+    namespaceSelector: {}
+    podSelector:
+      matchExpressions:
+        - {key: app, operator: NotIn, values: [authorized-client]}
+  egress:
+    - action: Deny
+      to:
+      - ipBlock:
+          cidr: 10.220.0.8/32  # restricted database running behind static IP
+```
+
+### Test Plan
+
+- Add e2e tests for ClusterNetworkPolicy resource
+  - Ensure `Deny` rules override all allowed traffic in the cluster
+  - Ensure `Allow` rules override K8s NetworkPolicies
+  - Ensure that in stacked ClusterNetworkPolicies/K8s NetworkPolicies, the following precedence is maintained
+    aggregated `Deny` rules > aggregated `Allow` rules > K8s NetworkPolicy rules
+- Add e2e tests for DefaultNetworkPolicy resource
+  - Ensure that in absence of ClusterNetworkPolicy rules and K8s NetworkPolicy rules, DefaultNetworkPolicy rules are observed
+  - Ensure that K8s NetworkPolicies override DefaultNetworkPolicies by applying policies to the same workloads
+  - Ensure that stacked DefaultNetworkPolicies are additive in nature
+- e2e test cases must cover ingress and egress rules
+- e2e test cases must cover port-ranges, named ports, integer ports etc
+- e2e test cases must cover various combinations of `podSelector` in `appliedTo` and ingress/egress rules
+- e2e test cases must cover various combinations of `namespaceSelector` in `appliedTo`
+- e2e test cases must cover various combinations of `namespaces` in ingress/egress rules
+  - Ensure that `except` field works as expected
+  - Ensure that `self` field works as expected
+- Add unit tests to test the validation logic which shall be introduced for cluster-scoped policy resources
+  - Ensure that `self` field cannot be set along with `selector` within `namespaces`
+  - Test cases for fields which are shared with NetworkPolicy, like `ipBlock`, `endPort` etc.
+- Ensure that only administrators or assigned roles can create/update/delete cluster-scoped policy resources
+
+### Graduation Criteria
+
+#### Alpha to Beta Graduation
+
+- Gather feedback from developers and surveys
+- At least 1 CNI provider must provide the implementation for the complete set
+  of alpha features
+- Evaluate "future work" items based on feedback from community
+
+#### Beta to GA Graduation
+
+- At least 2 CNI providers must provide the implementation for the complete set
+  of alpha features
```

Comment:
presumably should say "beta" there not alpha.

I'm not sure that "1" is a good value for the alpha-to-beta transition. I _am_ sure that "2" is not a good value for the beta-to-GA transition. This is a major feature. We want more buy-in than that before calling it GA.